app/models/service.rb

```diff
@@ -137,6 +137,14 @@ def power_status
     options[:power_status]
   end
 
+  def self.active
+    Service.where(:lifecycle_state => ["provisioning", "provisioned"], :retired => false)
+  end
+
+  def self.inactive
+    Service.where(:lifecycle_state => ["unprovisioned", "error_in_provisioning"], :retired => false).or(Service.retired)
+  end
```

Comment:
Simpler and also chainable:

```suggestion
  def self.active
    where(:lifecycle_state => ["provisioning", "provisioned"], :retired => false)
  end

  def self.inactive
    where(:lifecycle_state => ["unprovisioned", "error_in_provisioning"], :retired => false).or(retired)
  end
```